pkg/frontend/openshiftcluster_preflightvalidation.go

```diff
@@ -0,0 +1,109 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/database/cosmosdb"
+	"github.com/Azure/ARO-RP/pkg/frontend/middleware"
+)
+
+// NOTE: Make sure to change and always return 200, with any errors inside of the response message
+func (f *frontend) preflightValidation(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
+
+	var header http.Header
+	var b []byte
+
+	body := r.Context().Value(middleware.ContextKeyBody).([]byte)
+	correlationData := r.Context().Value(middleware.ContextKeyCorrelationData).(*api.CorrelationData)
+	systemData, _ := r.Context().Value(middleware.ContextKeySystemData).(*api.SystemData) // don't panic
+	originalPath := r.Context().Value(middleware.ContextKeyOriginalPath).(string)
+	referer := r.Header.Get("Referer")
+
+	subId := chi.URLParam(r, "subscriptionId")
+	resourceProviderNamespace := chi.URLParam(r, "resourceProviderNamespace")
+
+	apiVersion := r.URL.Query().Get(api.APIVersionKey)
+	err := cosmosdb.RetryOnPreconditionFailed(func() error {
```

Comment:
Why is `_preflightValidation` used in `cosmosdb.RetryonPreconditionFailed`? Is this for the following scenario:
1) customer creates a cluster
2) customer wants to update cluster config via ARM template
3) we run preflight validation on the new ARM template to ensure it's an OK update, and that validation may require looking at the cluster config in CosmosDB

Reply:
removed retryonPreconditionFailed